### PR TITLE
Update RandomLib to 1.1.0

### PR DESCRIPTION
**Pull request by LukasReschke**
Fixes https://github.com/owncloud/core/issues/14165 and a problem with HHVM. I also updated some files to the git ignore.

### CODE REVIEW

**Comment by LukasReschke** [comment]
@karlitschek Would love to backport this.

**Comment by LukasReschke** [comment]
@MorrisJobke @th3fallen Please test and review.

**Comment by LukasReschke** [comment]
Core: https://github.com/owncloud/core/pull/14167

**Comment by th3fallen** [comment]
:+1: Just a dep update.

**Comment by karlitschek** [comment]
@LukasReschke Hmm. A big backport. I´m ok with that if @DeepDiver1975 agrees too.

**Comment by LukasReschke** [comment]
It is not big. Those are effectively 10 LoC - rest is removed tests (added to .gitignore) and rewritten composer files ;-)

File: composer.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
 		"phpseclib/phpseclib": "*",
 		"rackspace/php-opencloud": "v1.9.2",
 		"james-heinrich/getid3": "dev-master",
-		"ircmaxell/random-lib": "v1.0.0",
+		"ircmaxell/random-lib": "~1.1",
```
Comment (DeepDiver1975) [inline composer.json]: hmmm - shouldn't we point to explicit versions? I'm still uncertain on how to do this "properly"

@bantu @nickvergessen

File: ircmaxell/random-lib/lib/RandomLib/Generator.php [hunk 2]
```diff
@@ -32,6 +32,66 @@
 class Generator {
 
     /**
+     * @const Flag for uppercase letters
```
Comment (DeepDiver1975) [inline ircmaxell/random-lib/lib/RandomLib/Generator.php]: @LukasReschke file permissions got changed - smells :fish:y

**Comment by DeepDiver1975** [comment]
Backport is fine for me on this - please submit a isolated PRs - once comments above got sorted out - THX

**Comment by DeepDiver1975** [comment]
:+1:

File: ircmaxell/random-lib/lib/RandomLib/Generator.php (hunk 2 above)
Comment (LukasReschke) [inline ircmaxell/random-lib/lib/RandomLib/Generator.php]: Mhm… They have the same value upstream as well. Let me fix this here and submit a pull request upstream as well.

File: composer.json (hunk 1 above)
Comment (nickvergessen) [inline composer.json]: This is the proper way.
The exact version is being locked in the composer.lock file.

We just have to take care, that on update people run `composer update vendor/package`, so they only update what they actually want to update.

File: .gitignore [hunk 3]
```diff
@@ -51,4 +51,14 @@ ircmaxell/password-compat/.travis.yml
...
-ircmaxell/password-compat/version-test.php
\ No newline at end of file
+ircmaxell/password-compat/version-test.php
+
+ircmaxell/random-lib/test/
+ircmaxell/random-lib/.gitignore
+ircmaxell/random-lib/.travis.yml
+ircmaxell/random-lib/LICENSE
+ircmaxell/random-lib/README.md
+ircmaxell/random-lib/composer.json
```
Comment (nickvergessen) [inline .gitignore]: not sure about this one

File: ircmaxell/random-lib/lib/RandomLib/Generator.php (hunk 2 above)
Comment (LukasReschke) [inline ircmaxell/random-lib/lib/RandomLib/Generator.php]: https://github.com/ircmaxell/RandomLib/pull/32

File: .gitignore (hunk 3 above)
Comment (LukasReschke) [inline .gitignore]: From what I can remember @MorrisJobke told me to also blacklist those files on some last PR here. I have no idea what is the right approach here :smile:

File: ircmaxell/random-lib/lib/RandomLib/Generator.php (hunk 2 above)
Comment (LukasReschke) [inline ircmaxell/random-lib/lib/RandomLib/Generator.php]: Reset to 644 as well here.

**Comment by LukasReschke** [comment]
Stable8: https://github.com/owncloud/3rdparty/pull/160 and https://github.com/owncloud/core/pull/14191

**Comment by PVince81** [comment]
Rebase required

**Comment by LukasReschke** [comment]
Rebased :see_no_evil:

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**